docs/DeveloperGuide.md

```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+    * 1a2 MediConnect requests for the correct input.
 
-  Use case ends.
+      Use case resumes at step 1.
 
-* 3a. The given index is invalid.
+* 1b. MediConnect detects that the appointment to be edited does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+  
+      Use case ends.
+* 3a.MediConnect detects that the arguments provided to edit an appointment is invalid.
+    * 3a1. MediConnect shows an error message.
+    * 3a2 MediConnect requests for the correct input
 
-    * 3a1. AddressBook shows an error message.
+      Use case resumes at step 3.
 
+**Use case: UC16 Delete an appointment**
+
+**MSS**
+
+1. User requests to list appointments 
+2. MediConnect shows a list of appointments
+3. User requests to delete an appointment in the list
+4. MediConnect deletes the appointment
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+    
+    Use case ends.
+* 3a. The given index is invalid.
+    * 3a1. MediConnect shows an error message.
       Use case resumes at step 2.
 
+**Use case: UC17 Find a doctor’s availabilities in a given timeframe**
+
+**MSS**
+
+1. User requests to find a doctor’s availability in a given timeframe.
+2. MediConnect shows the doctor’s availability.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a.The given command is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2 MediConnect requests for the correct input
+
+      Steps 1a1-1a2 are repeated until user inputs correct input.
+
+      Use case resumes at step 2
+
+* 1b. The given doctor does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+      Use case ends.
+
+**Use case: UC18 View a patient’s outstanding bills**
+
+**MSS**
+
+1. User requests to view a patient’s bills.
+2. MediConnect shows the patient’s bills.
+   Use case ends.
+
+**Extensions**
+
+* 1a.The given command is invalid.
+    * 1a1. MediConnect shows an error message. 
+    * 1a2 MediConnect requests for the correct input
+
+      Steps 1a1-1a2 are repeated until user inputs correct input.
+
+      Use case resumes at step 2
+
+* 1b. The given patient does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+      Use case ends.
+
+**Use case: UC19 Add an outstanding bill to a patient**
+
+**MSS**
+
+1. User requests to add an outstanding bill to a patient.
+2. MediConnect adds the outstanding bill to the patient.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a.The given command is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2 MediConnect requests for the correct input
+
+      Steps 1a1-1a2 are repeated until user inputs correct input.
+
+      Use case resumes at step 2
+
+* 1b. The given patient does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+      Use case ends.
+
+**Use case: UC20 Close an outstanding bill of a patient**
+
+**MSS**
+
+1. User requests to list outstanding bills a patient has
```

Comment:
@mounilsankar same thing should closing include listing. if so, is this supposed to refer back to the other use case instead?

Reply:
Yes, should not include listibng